decidim-surveys/spec/system/registered_user_survey_spec.rb

```diff
@@ -53,6 +53,75 @@
       login_as user, scope: :user
     end
 
+    # rubocop:disable Naming/VariableNumber
+    context "when survey allows editing" do
+      let(:question_description) do
+        {
+          "en" => "<p>Survey's content</p>",
+          "ca" => "<p>Contingut de l'enquesta</p>",
+          "es" => "<p>Contenido de la encuesta</p>"
+        }
+      end
+      let(:options) do
+        [
+          { "body" => Decidim::Faker::Localized.sentence },
+          { "body" => Decidim::Faker::Localized.sentence },
+          { "body" => Decidim::Faker::Localized.sentence }
+        ]
+      end
+      let(:max_characters) { 0 }
+      let(:max_choices) { nil }
+
+      let!(:survey) { create(:survey, :published, :allow_edit, :announcement, :allow_answers, :allow_unregistered, component:, questionnaire:) }
+      let!(:second_question) { create(:questionnaire_question, position: 1, questionnaire:, question_type: :multiple_option, max_choices:, max_characters:, options:) }
+      let!(:question) { create(:questionnaire_question, questionnaire:, mandatory: true, position: 0, description: question_description) }
+
+      before do
+        visit_component
+        click_on translated_attribute(questionnaire.title)
+
+        fill_in :questionnaire_responses_0, with: "My first answer"
+        check "questionnaire_tos_agreement"
+        accept_confirm { click_on "Submit" }
+      end
+
+      it "restricts the change of an answer when editing is disabled" do
+        expect(page).to have_content("Edit your answers")
+
+        survey.update!(allow_editing_answers: false)
+
+        click_on "Edit your answers"
+
+        expect(page).to have_content("You are not allowed to edit your answers.")
+      end
+
+      it "restricts the change of an answer when form is closed" do
+        expect(page).to have_content("Edit your answers")
+
+        survey.update!(ends_at: 1.day.ago)
+
+        click_on "Edit your answers"
+
+        expect(page).to have_content("You are not allowed to edit your answers.")
+      end
+
+      it "allows to change the response of a text field" do
+        expect(page).to have_content("Edit your answers")
+        click_on "Edit your answers"
+
+        expect(page).to have_field(:questionnaire_responses_0, with: "My first answer")
```

Comment:
Can you add more question types? As the data models are pretty different, I think we should cover all the cases.

Reply:
In 745701555b, i have added more questions types: 
- short_answer
- long_answer
- single_option
- multiple_option
- matrix_single
- matrix_multiple

The only ones not covered: sorting, files, separator